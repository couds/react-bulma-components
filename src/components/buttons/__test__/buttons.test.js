import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../../button';
import Buttons from '..';

describe('Buttons component', () => {
  it('Should exist', () => {
    expect(Buttons).toMatchSnapshot();
  });
  it('Should be a a default list of buttons', () => {
    const component = renderer.create(
      <Buttons>
        <Button>
          test
        </Button>
      </Buttons>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat class names in props with Bulma class name', () => {
    const component = renderer.create(
      <Buttons className="super-class-1 dope-class-2">
        <Button>
          test
        </Button>
      </Buttons>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should group buttons together', () => {
    const component = renderer.create(
      <Buttons hasAddons>
        <Button>
          test 1
        </Button>
        <Button>
          test 2
        </Button>
      </Buttons>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be centered', () => {
    const component = renderer.create(
      <Buttons position="centered">
        <Button>
          test
        </Button>
      </Buttons>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should align to the right', () => {
    const component = renderer.create(
      <Buttons position="right">
        <Button>
          test
        </Button>
      </Buttons>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
