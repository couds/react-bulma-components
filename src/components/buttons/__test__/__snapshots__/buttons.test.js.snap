// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons component Should align to the right 1`] = `
<div
  className="buttons is-right"
>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    tabIndex={0}
  >
    test
  </button>
</div>
`;

exports[`Buttons component Should be a a default list of buttons 1`] = `
<div
  className="buttons"
>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    tabIndex={0}
  >
    test
  </button>
</div>
`;

exports[`Buttons component Should be centered 1`] = `
<div
  className="buttons is-centered"
>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    tabIndex={0}
  >
    test
  </button>
</div>
`;

exports[`Buttons component Should concat class names in props with Bulma class name 1`] = `
<div
  className="buttons super-class-1 dope-class-2"
>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    tabIndex={0}
  >
    test
  </button>
</div>
`;

exports[`Buttons component Should exist 1`] = `[Function]`;

exports[`Buttons component Should group buttons together 1`] = `
<div
  className="buttons has-addons"
>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    tabIndex={0}
  >
    test 1
  </button>
  <button
    className="button"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    tabIndex={0}
  >
    test 2
  </button>
</div>
`;
